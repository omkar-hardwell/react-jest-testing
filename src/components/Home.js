import React from "react";
import { Link } from "react-router-dom";

const Home = () => {
  return <Link to="/employee">Employee view</Link>;
};

export default Home;
