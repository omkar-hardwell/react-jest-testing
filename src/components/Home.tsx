import React from "react";
import { Link } from "react-router-dom";

const Home = () => {
  return (
    <div className="ui container">
      <Link to="/employee">Employee view</Link>
    </div>
  );
};

export default Home;
