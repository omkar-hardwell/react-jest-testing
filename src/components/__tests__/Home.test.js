import { Router, MemoryRouter } from "react-router-dom";
import { act, create } from "react-test-renderer";
import { screen, render } from "@testing-library/react";
import { createMemoryHistory } from "history";

import Home from "../Home";

describe("Home page test with", () => {
  it("render a link", () => {
    let component;
    act(() => {
      component = create(
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      );
    });

    expect(component.toJSON()).toMatchSnapshot();
  });
  it("check valid link", () => {
    const history = createMemoryHistory();

    // mock push function
    history.push = jest.fn();

    render(
      <Router history={history}>
        <Home />
      </Router>
    );

    screen.getByText("Employee view");
    expect(screen.getByRole("link")).toHaveAttribute("href", "/employee");
  });
});
