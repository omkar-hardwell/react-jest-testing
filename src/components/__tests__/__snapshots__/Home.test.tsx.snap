// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page test with render a link 1`] = `
<div
  className="ui container"
>
  <a
    href="/employee"
    onClick={[Function]}
  >
    Employee view
  </a>
</div>
`;
